lib/portlayer/storage/vsphere/image.go

```diff
@@ -641,3 +661,61 @@ func imagesInUse(op trace.Operation, ID string) error {
 
 	return nil
 }
+
+// populate the scratch with minimum OS structure defined in FileForMinOS and DirForMinOS
+func populateMinOS(op trace.Operation, vmdisk *disk.VirtualDisk) error {
+	// tmp dir to mount the disk
+	dir, err := ioutil.TempDir("", "mnt-"+portlayer.Scratch.ID)
+	if err != nil {
+		op.Errorf("Failed to create tempDir: %s", err)
+	}
+	defer os.RemoveAll(dir)
+
+	if err = vmdisk.Mount(dir, nil); err != nil {
+		op.Errorf("Failed to mount device %s to dir %s", vmdisk.DevicePath, dir)
+	}
+
+	for dname, dmode := range DirForMinOS {
+		dirPath := fmt.Sprintf("%s%s", dir, dname)
+		m, err := strconv.ParseUint(dmode, 0, 0)
+		if err != nil {
+			op.Errorf("Failed to ParseUint for %s: %e", dmode, err)
+		}
+		if err = os.MkdirAll(dirPath, os.FileMode(m)); err != nil {
+			op.Errorf("Failed to create directory %s: %s", dirPath, err)
+			return err
+		}
+	}
+
+	// The directory has to exist before creating the new file
+	for fname, fmode := range FileForMinOS {
+		filePath := fmt.Sprintf("%s%s", dir, fname)
+		f, err := os.Create(filePath)
```

Comment:
If you use the following, along with the mode suggestion for the directory map then this loop becomes much smaller:
`os.OpenFile(filepath, os.O_CREATE, mode)`